recipes/ggd/meta.yaml

```
@@ -0,0 +1,54 @@
+{% set version = "0.1.0" %}
+
+package:
+  name: ggd
+  version: {{ version }}
+
+source:
+  url: https://github.com/gogetdata/ggd-cli/archive/v{{ version }}.tar.gz
+  sha256: 540c67b4beb9dee34fa07c55e86e6f58c17dbc345a4700207cbc87615040a7ff
+
+build:
+    number: 0
+    noarch: python
+    entry_points:
+      - ggd = ggd.__main__:main
+    script: "python -m pip install . --no-deps --ignore-installed -vv" 
+
+requirements:
+  host:
+    - pip
+    - pyyaml
```

```suggestion
    - python
```
We should have both `pip` and `python` here since we run `python -m pip`, i.e., the build directly depends on both (better be explicit in this case).
(Plus, I don't know whether `conda-build` does some funky `python`-specific things that expect `requirements/host/python` to be present.)
We can remove `pyyaml` since with `pip install --no-deps` the build won't touch `pyyaml` at all now, which makes things easier.

---

When python is added to the meta.yaml, as suggested above, it installs python2 in a python3 environment. GGD works for both python2 and 3 so I can't designate which python version to install. I'm not sure why this is going on, but I would rather not change the base python version when installing GGD. Any suggestions?

---

Hm, sound like a coincidence, maybe? With the `build/noarch: python` specified, we should get a package that just depends on `python` with no additional version constraints, so I'm a bit surprised about what you described.
Can you change the recipe in this PR as suggested with `python` both in `host` and `run`? then I can inspect the build log and created package to check what's happening.

---

meta.yaml is changed.

---

The build log is consistent with what I see on my local machine, that is, python 2 is being installed when `- python` is set in `build/host` even when you are using a python 3 environment.

---

I'm not sure where you see that in the logs, conda just randomly chooses different python versions for different independent steps in cases like this, which is perfectly fine.

---

bioconda-utils depends on python3, which I assume is installed because bioconda-utils is running. From the log:
```
16:00:43 BIOCONDA INFO (OUT)     _libgcc_mutex:   0.1-main             defaults
16:00:43 BIOCONDA INFO (OUT)     ca-certificates: 2019.9.11-hecc5488_0 conda-forge
16:00:43 BIOCONDA INFO (OUT)     certifi:         2019.9.11-py27_0     conda-forge
16:00:43 BIOCONDA INFO (OUT)     libffi:          3.2.1-he1b5a44_1006  conda-forge
16:00:43 BIOCONDA INFO (OUT)     libgcc-ng:       9.1.0-hdf63c60_0     defaults
16:00:43 BIOCONDA INFO (OUT)     libstdcxx-ng:    9.1.0-hdf63c60_0     defaults
16:00:43 BIOCONDA INFO (OUT)     ncurses:         6.1-hf484d3e_1002    conda-forge
16:00:43 BIOCONDA INFO (OUT)     openssl:         1.1.1d-h516909a_0    conda-forge
16:00:43 BIOCONDA INFO (OUT)     pip:             19.3.1-py27_0        conda-forge
16:00:43 BIOCONDA INFO (OUT)     python:          2.7.15-h5a48372_1009 conda-forge
16:00:43 BIOCONDA INFO (OUT)     readline:        8.0-hf8c457e_0       conda-forge
16:00:43 BIOCONDA INFO (OUT)     setuptools:      41.6.0-py27_1        conda-forge
16:00:43 BIOCONDA INFO (OUT)     sqlite:          3.30.1-hcee41ef_0    conda-forge
16:00:43 BIOCONDA INFO (OUT)     tk:              8.6.9-hed695b0_1003  conda-forge
16:00:43 BIOCONDA INFO (OUT)     wheel:           0.33.6-py27_0        conda-forge
16:00:43 BIOCONDA INFO (OUT)     zlib:            1.2.11-h516909a_1006 conda-forge
```

Note that python 2 is designated. 

A warning from pip, suggesting it is the version being used: 
```
16:01:07 BIOCONDA INFO (OUT) DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
```

---

You're right, but everything is fine ;).
- `bioconda-utils`+`python=3` is used as the base for the CI.
- The build via `conda-build` happens in a container that also has `bioconda-utils`+`python=3`
- `conda-build` chooses whatever it likes (within the given constraints, of course) during the build. Very often it chooses `python=2` (possibly due to the order in which possible `python` versions are given in our build setup). This is a little strange, admittedly, but does not matter since `conda-build`+`pip` usually build the same `noarch: python` package, no matter which `python` version is used during the build.
- The built package is tested in an additional container at the end of the CI. You can see that it chooses `python=3.7` there, i.e., we've built with `python=2` but the package tests fine in a `python=3` environment.